File: web/src/types/log.ts
// 日志级别枚举
export enum LogLevel {
  DEBUG = 'debug',
  INFO = 'info',
  WARN = 'warn',
  ERROR = 'error',
  FATAL = 'fatal',
}

// 日志类别枚举
export enum LogCategory {
  SYSTEM = 'system',
  USER = 'user',
  AUTH = 'auth',
  API = 'api',
  MODEL = 'model',
  DATABASE = 'database',
  PERFORMANCE = 'performance',
}

// 日志类型枚举
export enum LogType {
  ALL = 'all',
  API = 'api',
  ERROR = 'error',
  MODEL_CALL = 'model_call',
}

// 用户信息接口
export interface User {
  id: string;
  username: string;
  email: string;
}

// 基础日志接口
export interface Log {
  id: string;
  level: LogLevel;
  category: LogCategory;
  message: string;
  user_id?: string;
  user?: User;
  metadata?: any;
  created_at: string;
}

// API日志接口
export interface APILog extends Log {
  method: string;
  path: string;
  status_code: number;
  ip?: string;
  user_agent?: string;
  duration: number;
  request_id?: string;
}

// 错误日志接口
export interface ErrorLog extends Log {
  stack_trace?: string;
  error_code?: string;
  source?: string;
  resolved_at?: string;
  resolved_by?: string;
}

// 模型调用日志接口
export interface ModelCallLog extends Log {
  model_name: string;
  prompt_tokens?: number;
  comp_tokens?: number;
  total_tokens?: number;
  duration: number;
  application_id?: string;
  project_id?: string;
  success: boolean;
}

// 系统指标接口
export interface SystemMetric {
  id: string;
  metric_name: string;
  metric_value: number;
  unit?: string;
  tags?: any;
  created_at: string;
}

// 日志响应接口，用于统一处理后端返回的日志数据
export interface LogResponse {
  data: Log | APILog | ErrorLog | ModelCallLog;
}

// 日志列表响应接口
export interface LogListResponse {
  data: (Log | APILog | ErrorLog | ModelCallLog)[];
  meta: {
    total: number;
    page: number;
    size: number;
  };
}

// 日志统计响应接口
export interface LogStatsResponse {
  data: {
    total_logs: number;
    error_count: number;
    api_count: number;
    model_call_count: number;
    avg_api_duration: number;
    avg_model_call_duration: number;
    error_by_category: Record<string, number>;
    logs_by_level: Record<string, number>;
  };
  meta: {
    start_time: string;
    end_time: string;
  };
}

// 系统指标响应接口
export interface MetricsResponse {
  data: {
    cpu_usage: SystemMetric[];
    memory_usage: SystemMetric[];
    disk_usage: SystemMetric[];
    network_traffic: SystemMetric[];
    api_latency: SystemMetric[];
    model_latency: SystemMetric[];
  };
} 

File: server/models/log.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

// LogLevel 日志级别枚举
type LogLevel string

const (
	// LogLevelDebug 调试级别
	LogLevelDebug LogLevel = "debug"
	// LogLevelInfo 信息级别
	LogLevelInfo LogLevel = "info"
	// LogLevelWarn 警告级别
	LogLevelWarn LogLevel = "warn"
	// LogLevelError 错误级别
	LogLevelError LogLevel = "error"
	// LogLevelFatal 致命级别
	LogLevelFatal LogLevel = "fatal"
)

// LogCategory 日志类别枚举
type LogCategory string

const (
	// LogCategorySystem 系统日志
	LogCategorySystem LogCategory = "system"
	// LogCategoryAPI API请求日志
	LogCategoryAPI LogCategory = "api"
	// LogCategoryAuth 认证日志
	LogCategoryAuth LogCategory = "auth"
	// LogCategoryModel 模型调用日志
	LogCategoryModel LogCategory = "model"
	// LogCategoryApplication 应用日志
	LogCategoryApplication LogCategory = "application"
)

// Log 基础日志模型
type Log struct {
	ID        uuid.UUID   `gorm:"type:uuid;primary_key" json:"id"`
	Level     LogLevel    `gorm:"type:varchar(10);not null;index" json:"level"`
	Category  LogCategory `gorm:"type:varchar(20);not null;index" json:"category"`
	Message   string      `gorm:"type:text;not null" json:"message"`
	UserID    *uuid.UUID  `gorm:"type:uuid;index" json:"user_id,omitempty"`
	User      *User       `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Metadata  string      `gorm:"type:jsonb" json:"metadata"`
	CreatedAt time.Time   `json:"created_at"`
}

// BeforeCreate 在创建日志前生成UUID
func (l *Log) BeforeCreate(tx *gorm.DB) error {
	l.ID = uuid.New()
	return nil
}

// APILog API日志模型，扩展自Log
type APILog struct {
	Log
	Method     string `gorm:"type:varchar(10);not null" json:"method"`
	Path       string `gorm:"type:varchar(255);not null" json:"path"`
	StatusCode int    `gorm:"not null" json:"status_code"`
	IP         string `gorm:"type:varchar(50)" json:"ip"`
	UserAgent  string `gorm:"type:varchar(255)" json:"user_agent"`
	Duration   int64  `gorm:"not null" json:"duration"` // 毫秒
	RequestID  string `gorm:"type:varchar(36);index" json:"request_id"`
}

// ErrorLog 错误日志模型，扩展自Log
type ErrorLog struct {
	Log
	StackTrace string    `gorm:"type:text" json:"stack_trace"`
	ErrorCode  string    `gorm:"type:varchar(50)" json:"error_code"`
	Source     string    `gorm:"type:varchar(100)" json:"source"`
	ResolvedAt *time.Time `json:"resolved_at,omitempty"`
	ResolvedBy *uuid.UUID `gorm:"type:uuid" json:"resolved_by,omitempty"`
}

// ModelCallLog 模型调用日志，记录LLM调用
type ModelCallLog struct {
	Log
	ModelName    string    `gorm:"type:varchar(100);not null;index" json:"model_name"`
	PromptTokens int       `json:"prompt_tokens"`
	CompTokens   int       `json:"comp_tokens"`
	TotalTokens  int       `json:"total_tokens"`
	Duration     int64     `gorm:"not null" json:"duration"` // 毫秒
	ApplicationID *uuid.UUID `gorm:"type:uuid;index" json:"application_id,omitempty"`
	ProjectID    *uuid.UUID `gorm:"type:uuid;index" json:"project_id,omitempty"`
	Success      bool      `gorm:"not null" json:"success"`
}

// SystemMetric 系统指标记录
type SystemMetric struct {
	ID          uuid.UUID `gorm:"type:uuid;primary_key" json:"id"`
	MetricName  string    `gorm:"type:varchar(100);not null;index" json:"metric_name"`
	MetricValue float64   `gorm:"not null" json:"metric_value"`
	Unit        string    `gorm:"type:varchar(20)" json:"unit"`
	Tags        string    `gorm:"type:jsonb" json:"tags"` // JSON格式的标签
	CreatedAt   time.Time `json:"created_at"`
}

// BeforeCreate 在创建系统指标前生成UUID
func (sm *SystemMetric) BeforeCreate(tx *gorm.DB) error {
	sm.ID = uuid.New()
	return nil
}

// LogResponse 是返回给客户端的日志数据结构
type LogResponse struct {
	ID        uuid.UUID   `json:"id"`
	Level     LogLevel    `json:"level"`
	Category  LogCategory `json:"category"`
	Message   string      `json:"message"`
	UserID    *uuid.UUID  `json:"user_id,omitempty"`
	UserName  string      `json:"user_name,omitempty"`
	Metadata  interface{} `json:"metadata,omitempty"`
	CreatedAt time.Time   `json:"created_at"`
	// 扩展字段，根据日志类型填充
	Method      string     `json:"method,omitempty"`
	Path        string     `json:"path,omitempty"`
	StatusCode  int        `json:"status_code,omitempty"`
	Duration    int64      `json:"duration,omitempty"`
	ModelName   string     `json:"model_name,omitempty"`
	TotalTokens int        `json:"total_tokens,omitempty"`
	ErrorCode   string     `json:"error_code,omitempty"`
	ResolvedAt  *time.Time `json:"resolved_at,omitempty"`
} 